### ISSUE TITLE
Missing headers, params and query

### ISSUE THREAD

**Issue by rama41222** [knaxus/nodejs-api-gateway#1]
#### Proxying headers, params and query

The API Gateway should handle headers, params and query params accordingly.

**Comment by ashokdey** [comment]
O shit!

**Comment by ashokdey** [comment]
@rama41222, I added the missing entities this way:

```javascript
const apiOptions = {
    method: req.method,
    url: req.originalUrl,
    query: req.query,
    params: req.params,
    headers: req.headers,
    body: req.body,
    baseURL: BASE_URL_BOOK,
  };
```

But now I am thinking whether I can clone the request and pass it forward. Is there any mechanism for the same that you have tried? I have to read about it.

**Comment by rama41222** [comment]
yeah I also wanted to do the similar thing. to extract out all the details from the request and pipe it to the service. I found this package, May be you can have a look at implementation details etc
[node-http-proxy](https://github.com/nodejitsu/node-http-proxy)
